### PR TITLE
Remove our fork of forkjoin. Java 8 bundles it.

### PR DESCRIPTION
**Pull request by adriaanm**
Provide deprecated compatibility stubs for the types and static members,
which forward as follows:

```
scala.concurrent.forkjoin.ForkJoinPool         => java.util.concurrent.ForkJoinPool
scala.concurrent.forkjoin.ForkJoinTask         => java.util.concurrent.ForkJoinTask
scala.concurrent.forkjoin.ForkJoinWorkerThread => java.util.concurrent.ForkJoinWorkerThread
scala.concurrent.forkjoin.LinkedTransferQueue  => java.util.concurrent.LinkedTransferQueue
scala.concurrent.forkjoin.RecursiveAction      => java.util.concurrent.RecursiveAction
scala.concurrent.forkjoin.RecursiveTask        => java.util.concurrent.RecursiveTask
scala.concurrent.forkjoin.ThreadLocalRandom    => java.util.concurrent.ThreadLocalRandom
```

To prepare for Java 9, the Scala library avoids using `sun.misc.Unsafe`.
However, for now, it does provide a convenience accessor for it via `scala.concurrent.util.Unsafe`.
This (deprecated) class will be removed as soon as the eco-system drops its use
(akka-actor, I'm looking at you -- cc @viktorklang).

### CODE REVIEW

**Comment by adriaanm** [comment]
Community build :green_heart: at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/72

**Comment by adriaanm** [comment]
Review by @SethTisue

**Comment by VladUreche** [comment]
This patch will make the Oracle guys working on Project Jigsaw very happy, as it removes the Unsafe dependency :+1: 

https://issues.scala-lang.org/browse/SI-9381

**Comment by viktorklang** [comment]
Thanks @ktoso for https://github.com/akka/akka/pull/18009

**Comment by retronym** [comment]
@adriaanm I've removed further detritus here: https://github.com/adriaanm/scala/compare/adriaanm:unforkjoin...retronym:review/unforkjoin

**Comment by adriaanm** [comment]
Thanks! Looks like I missed quite a few spots :-) Squashed them onto my first round of sweeping.

**Comment by retronym** [comment]
Not sure quite how to manage the updates for these, but I've grepped for `forkjoin` in the default branch of a few GitHub orgs to show the spots that ought be updated eventually: https://gist.github.com/retronym/979da8069e04fd14b448

Maybe we can find some additional defacto API that needs to be stubbed in these, too.

**Comment by retronym** [comment]
LGTM, I've added the "release-notes" label.

**Comment by retronym** [comment]
LGTM

**Comment by SethTisue** [comment]
@axel22 and @Ichoran, the removal of benchmark code might interest you

File: src/library/scala/concurrent/util/Unsafe.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package scala.concurrent.util;
+import java.lang.reflect.Field;
+
+// TODO: remove once akka no longer needs it, hopefully by 2.12.0-M3!
+@Deprecated
+public final class Unsafe {
```
Comment (axel22) [inline src/library/scala/concurrent/util/Unsafe.java:14]: There are a couple of other places where we'll need to get rid of `Unsafe`, most notably, `Future`s and `TrieMap`.

**Comment by axel22** [comment]
I'm a little bit surprised that we decided to remove the old (currently unused) benchmark code as part of this PR (which is about dropping our bundled Fork/Join pools) - I thought that PRs are supposed to be on-topic. But that's ok with me.

For what it's worth, I would be interested in reviving some of these benchmarks, and making sure that they run on a nightly basis, along with producing some [fancy reports](http://scala-blitz.github.io/home/documentation/benchmarks//chara.html). These need not impact the correctness of the build, but would be useful as a sanity check - we [often ask our submitters to do some benchmarks](https://github.com/scala/scala/pull/4504). We certainly have [the infrastructure to do that systematically now](http://scalameter.github.io/).

File: src/library/scala/concurrent/util/Unsafe.java (hunk 1 above)
Comment (viktorklang) [inline src/library/scala/concurrent/util/Unsafe.java:14]: Futures don't use Unsafe anymore AFAIK

Comment (axel22) [inline src/library/scala/concurrent/util/Unsafe.java:14]: One left to fix then.

Comment (axel22) [inline src/library/scala/concurrent/util/Unsafe.java:14]: Correction - the `TrieMap` uses `AtomicReferenceFieldUpdater`s - we never changed it to `Unsafe`. So, we're good there too.

**Pull request merged**